### PR TITLE
LPS-67491 Settings framework doesn't respect Meta.AD annotation id property anymore

### PR DESCRIPTION
**Pull request by drewbrokke**
This is an update for [LPS-67491](https://issues.liferay.com/browse/LPS-67491).<br><br>Hey @mikakoivisto, this change will check the "id" attribute value first, and if the result is null will then perform the same check with the method name.  I had to duplicate the return-type checking logic because I had to call `_typedSettings.getValue(key, null)` with null as the default value so we can test for it (otherwise we'd receive the `get[sometype]Value` default).  Let me know if this works.  Thanks! :)<br><br>/cc @jorgeferrer @pei-jung

### CODE REVIEW

**Comment by drewbrokke** [comment]
ci:retest

File: modules/apps/foundation/portal-configuration/portal-configuration-module-configuration/src/main/java/com/liferay/portal/configuration/module/configuration/internal/ConfigurationInvocationHandler.java [hunk 1]
```diff
@@ -65,61 +67,122 @@ public Object invoke(Object proxy, Method method, Object[] args)
 		}
 	}
 
-	private Object _invokeConfigurationOverride(Method method, Object[] args)
-		throws IllegalAccessException, InvocationTargetException,
-			   NoSuchMethodException {
+	private Object _getValueNoDefault(Class<?> returnType, String key)
```
Comment (mikakoivisto) [inline modules/apps/foundation/portal-configuration/portal-configuration-module-configuration/src/main/java/com/liferay/portal/configuration/module/configuration/internal/ConfigurationInvocationHandler.java]: This method doesn't seem right to me as it always returns a string type value. Also I'd compress _getValueNoDefault and _getValueWithDefault to just a single method with a boolean for whether to use the 2 arg version of the method or not.

**Comment by mikakoivisto** [comment]
Drew can you modify it based on my comments here https://github.com/mikakoivisto/liferay-portal/pull/132/commits/d306c38c70922ee87df5d9b7bf26b175525fe492#diff-0677b887cd564737386191dba2e60db5R70

**Comment by drewbrokke** [comment]
New pull request sent: mikakoivisto#133

**Pull request closed without merging**